app/src/main/java/me/quadphase/qpdex/AdvancedSearch.java

```diff
@@ -0,0 +1,508 @@
+package me.quadphase.qpdex;
+
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Spinner;
+import android.content.Intent;
+
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import me.quadphase.qpdex.databaseAccess.PokemonFactory;
+import me.quadphase.qpdex.pokemon.Ability;
+import me.quadphase.qpdex.pokemon.EggGroup;
+import me.quadphase.qpdex.pokemon.MinimalPokemon;
+import me.quadphase.qpdex.pokemon.Type;
+
+
+/**
+ * Created by Thinesh on 31-Aug-15
+ *
+ * Class deals with the advanced search feature accessed from main menu
+ *
+ * Can filter pokemon by
+ *
+ * Type 1
+ * Type 2
+ * Ability
+ * Generation Available
+ * Egg Groups 1
+ * Egg Groups 2
+ * TODO: All 6 Base stats
+ * TODO: The ability to include/exclude Primal/Mega/Alternate Forms
+ *
+ */
+
+
+public class AdvancedSearch extends ActionBarActivity {
+
+
+    // pokemonFactory to access the various DB access functions
+    PokemonFactory pokemonFactory;
+
+    // Pokemon information to filter
+    MinimalPokemon[] allMinimalPokemon;
+
+    //Final Filtered List
+    MinimalPokemon [] filteredMinimalPokemon;
+    ArrayList<Integer> filteredNationalIds;
+    ArrayList<Integer> filteredUniqueIds;
+    //Contains either or
+    ArrayList<Integer> currentFilterPokemonArrayList;
+
+    // Arrays to fill Spinners
+    Type[] allTypes;
+    Ability[] allAbilities;
+    EggGroup[] allEggGroups;
+    String allGenerations [];

```

Comment:
According to [stackoverflow](http://stackoverflow.com/questions/1200621/declare-array-in-java) both work, but [] after String is preferred. So your choice.